src/Resolvers/DataFromArrayResolver.php

```diff
@@ -91,14 +91,23 @@ protected function createData(
             }
         }
 
-        try {
-            return new $dataClass->name(...$parameters);
-        } catch (ArgumentCountError $error) {
+        if ($this->isAnyParameterMissing($dataClass, array_keys($parameters))) {
             throw CannotCreateData::constructorMissingParameters(
                 $dataClass,
                 $parameters,
-                $error
             );
         }
+
+        return new $dataClass->name(...$parameters);
+    }
+
+    protected function isAnyParameterMissing(DataClass $dataClass, array $parameters): bool
+    {
+        return $dataClass
+            ->constructorMethod
+            ->parameters
+            ->pluck('name')
```

Comment:
```suggestion
            ->filter(fn ($p) => !$parameter->hasDefaultValue)
            ->pluck('name')
```

Reply:
> Wouldn't this filter out required parameters as well, which do not have a default value?

It would keep only parameters without default values aka required ones, though there is one obscure area, is if you have a default value before a required param (which normally emits a warning)

Which is why filtering on is optional would provide the perfect solution, but it would need to be added to the DataParameter class and factory..
Or a new Reflection needs to be made on the spot